queries/robot/highlights.scm

```diff
@@ -1,21 +1,58 @@
-(argument (dictionary_variable) @string.special)
-(argument (list_variable) @string.special)
-(argument (scalar_variable) @string.special)
-(argument (text_chunk) @string)
+[
+  (comment)
+  (extra_text)
+] @comment
 
-(keyword_invocation (keyword) @function)
+[
+  (section_header)
+  (setting_statement)
+] @keyword
 
-(test_case_definition (name) @property)
+(variable_definition (variable_name) @variable)
 
-(keyword_definition (body (keyword_setting) @keyword))
 (keyword_definition (name) @function)
+(keyword_definition (body (keyword_setting) @keyword))
 
-(variable_definition (variable_name) @variable)
+(test_case_definition (name) @function)
+(test_case_definition (body (test_case_setting) @keyword))
```

Comment:
Same question, can these two keyword nodes be keyword in every case or does it need to be this specific? If not, keyword_setting and test_case_setting can be moved to the chunk above with other keywords

Reply:
Those nodes can only occur in this location, so it felt natural to capture it like this. I've changed it now.

Reply:
It's easier to group them together as a "keyword" block if they can only be keywords :)

See my other comment about FOR/END etc, your old style is totally okay too (Lua does this)